File: 274. H-Index/274. H-Index.js
/**
 * @param {number[]} citations
 * @return {number}
 */
var hIndex = function (citations) {
  const n = citations.length;
  const numCitations = new Array(n + 1).fill(0);

  for (const c of citations) {
    numCitations[Math.min(c, n)]++;
  }

  let papers = 0;
  for (let h = n; h > -1; h--) {
    papers += numCitations[h];
    if (papers >= h) {
      return h;
    }
  }
};

File: 274. H-Index.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        n = len(citations)
        citations_count = [0] * (n + 1)
        for c in citations:
            if c >= n:
                citations_count[n] += 1
            else:
                citations_count[c] += 1

        papers = 0
        for h in range(n, -1, -1):
            papers += citations_count[h]
            if papers >= h:
                return h
